extensions/cornerstone/src/components/ViewportWindowLevel/ViewportWindowLevel.tsx

```diff
@@ -0,0 +1,329 @@
+    const value = data[index];
+    if (value < minToUse) continue;
+    if (value > maxToUse) continue;
+    const bin = Math.floor((value - minToUse) * binScale);
+    bins[bin] += 1;
+  }
+
+  histogram.bins = bins;
+  histogram.maxBin = 0;
+  histogram.maxBinValue = 0;
+
+  for (let bin = 0; bin < histogram.numBins; bin++) {
+    if (histogram.bins[bin] > histogram.maxBinValue) {
+      histogram.maxBin = bin;
+      histogram.maxBinValue = histogram.bins[bin];
+    }
+  }
+
+  return histogram;
+};
+
+const ViewportWindowLevel = ({
+  servicesManager,
+  viewportIndex,
+}: {
+  servicesManager: ServicesManager;
+  viewportIndex: number;
+}): ReactElement => {
+  const { cornerstoneViewportService } = servicesManager.services;
+  const [windowLevels, setWindowLevels] = useState([]);
+
+  const getViewportVolumeHistogram = useCallback(
+    (viewport, volume, options?) => {
+      if (!volume?.loadStatus.loaded) {
+        return undefined;
+      }
+
+      const volumeImageData = viewport.getImageData(volume.volumeId);
+
+      if (!volumeImageData) {
+        return undefined;
+      }
+
+      const { scalarData, imageData } = volumeImageData;
+      const range = imageData.computeHistogram(imageData.getBounds());
+      const { minimum: min, maximum: max } = range;
+      const calcHistOptions = {
+        numBins: 256,
+        min: Math.max(min, options?.min ?? min),
+        max: Math.min(max, options?.max ?? max),
+      };
+
+      return calcHistogram(scalarData, calcHistOptions);
+    },
+    []
+  );
+
+  const getWindowLevelsData = useCallback(
+    (viewportIndex, prevWindowLevels = []) => {
+      const viewport = cornerstoneViewportService.getCornerstoneViewportByIndex(
+        viewportIndex
+      );
+
+      if (!viewport) {
+        return [];
+      }
+
+      const viewportInfo = cornerstoneViewportService.getViewportInfoByIndex(
+        viewportIndex
+      );
+
+      const volumeIds = viewport.getActors().map(actor => actor.uid);
+      const viewportProperties = viewport.getProperties();
+      const { voiRange } = viewportProperties;
+      const viewportVoi = voiRange
+        ? {
+            windowWidth: voiRange.upper - voiRange.lower,
+            windowCenter:
+              voiRange.lower + (voiRange.upper - voiRange.lower) / 2,
+          }
+        : undefined;
+
+      const windowLevels = volumeIds
+        .map((volumeId, volumeIndex) => {
+          const prevWindowLevel = prevWindowLevels.find(
+            prev => prev.volumeId === volumeId
+          );
+
+          if (prevWindowLevel) {
+            return { ...prevWindowLevel };
+          }
+
+          const volume = cs3DCache.getVolume(volumeId);
+          const { metadata, scaling } = volume;
+          const modality = metadata.Modality;
+
+          // TODO: find a proper way to fix the histogram
+          const options = {
+            min: modality === 'PT' ? 0.1 : -999,
+            max: modality === 'PT' ? 5 : 1000,
+          };
```

Comment:
CT goes to >5k and PT > 156 pixel values. Looking for a way to fix the histogram